app/Views/Templates/components/forms/text-input.blade.php

```diff
@@ -0,0 +1,48 @@
+@props([
+    'inputType' => 'text',
+    'id' => '',
+    'name' => '',
+    'placeholder' => '',
+    'label' => '',
+    'size' => '',
+    'inputState' => '',
+    'value' => '',
+    'leadingVisual' => '',
+    'trailingVisual' => '',
+])
+
+@php
+    $sizeClass = $size ? 'input-'.$size : '';
+    $stateClass = $inputState ? 'input-'.$inputState : '';
+@endphp
+
+<div class='par relative w-full max-w-xs'>
+    @if($label)
+        <label for="{{ $id }}">
+            <span class="label-text">{{ $label }}</span>
+        </label>
+    @endif
+
+    <div class="relative">
+        @if($leadingVisual)
+            <span class="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
+                <i class="{{ $leadingVisual }}"></i>
```

Comment:
I like it but more than likely we will be using a different icon library moving forward (blade ui kit). Which uses blade component styles or svg directives. I think we'd want to be flexible here and just display the "leadingvisual" without being forced to be a class on an i element